src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -275,9 +275,12 @@
             (when timestamp [:div {} "Occurred: " (-> timestamp js/moment (.format "LLL Z"))])
             (when source [:div {} "Source: " source])
             (when (seq causes)
-              [:div {}
-               [:div {} (str "Cause" (when (> (count causes) 1) "s") ":")]
-               (map (fn [cause] [CauseViewer cause]) causes)])
+              (let [num-hidden (- (count causes) 4)]
+                [:div {}
+                 [:div {} (str "Cause" (when (> (count causes) 1) "s") ":")]
+                 (map (fn [cause] [CauseViewer cause]) (take 4 causes))
+                 (when (pos? num-hidden)
+                   [:div {} (str num-hidden " not shown")])]))
```

Comment:
What's happening here? Does the ErrorViewer now show a max of 4 causes? I don't necessarily object to this, but it feels ... unrelated. Can you talk about why this is necessary?